docs/spec/src/integration/spec/5-lifecycle-phases.md

```diff
@@ -0,0 +1,169 @@
+
+```python
+
+class CertV2:
+    batch_header: any  # You can specify proper types here
+    blob_verification_proof: any
+    nonsigner_stake_sigs: any
+
+def get_cert_v2(blob_header_hash, blob_verifier_binding) -> CertV2:
+    """
+    V2 cert construction pseudocode
+    @param blob_header_hash: key used for referencing blob status from disperser
+    @param blob_verifier_binding: ABI contract binding used for generating nonsigner metadata
+    @return v2_cert: EigenDA V2 certificate used by rollup 
+    """
+  # Call the disperser for the info needed to construct the cert
+    blob_status_reply = disperser_client.get_blob_status(blob_header_hash)
+    
+    # Validate the blob_header received, since it uniquely identifies
+    # an EigenDA dispersal.
+    blob_header_hash_from_reply = blob_status_reply
+                                                                .blob_verification_info
+                                                                .blob_certificate
+                                                                .blob_header
+                                                                .Hash()
+    if blob_header_hash != blob_header_hash_from_reply {
+        throw/raise/panic
+    }
+
+    # Extract first 2 cert fields from blob status reply
+    batch_header = blob_status_reply.signed_batch.batch_header
+    blob_verification_proof = blob_status_reply.blob_verification_info
+    
+    # Construct NonSignerStakesAndSignature
+    nonsigner_stake_sigs = blob_verifier_binding.getNonSignerStakesAndSignature(
+                                                     blob_status_reply.signed_batch)
+                                                 
+  return Cert(batch_header, blob_verification_proof, nonsigner_stake_sigs)
+```
+
+## Posting to Ethereum
+
+The proxy converts the `cert` to an [`altda-commitment`](./3-datastructs.md#altdacommitment) ready to be submitted to the batcher’s inbox without any further modifications by the rollup stack.
+
+## Retrieval
+
+There are two main blob retrieval paths:
+
+1. decentralized trustless retrieval: retrieve chunks from Validators are recreate the `blob` from them.
+2. centralized trustful retrieval: the same [Relay API](https://docs.eigenda.xyz/releases/v2#relay-interfaces) that Validators use to download chunks, can also be used to retrieve full blobs.
```

Comment:
>trustless retrieval
> trustful retrieval

IMO both of these retrieval methods can be considered trustless, in that it's not possible in either case to provide undetectable false information. I would just remove the words "trustless" and "trustful" from these descriptions, to avoid confusion

Reply:
the trust is for liveness; not safety

Reply:
but agree its not needed; will remove

Reply:
[176dd65](https://github.com/Layr-Labs/eigenda/pull/1439/commits/176dd652e57e14075b2e32d93af34488a4fff2d9)